### DISCUSSION TITLE
Allow field hooks to run onChange

### DISCUSSION THREAD

**Discussion by annardini** [payloadcms/payload#2666] [Feature Requests & Ideas]
I am trying to use a relationship field that when data is changed (`onChange`), automatically updates some read-only fields in an array in my collection. This is to prevent user error when trying to add them manually. However, Payload hooks only seem to run `onSubmit` so the user would have to save then go back and edit again which is not good UX. Haven't been able to find a work around yet but think it would be good to implement for at least some hooks like `beforeChange`, `afterChange`, `beforeValidate` and `afterRead`.

**Comment by rhysmakesthings** [comment]
+1. I'm having a similar issue! I'm using dispatchField "ADD_ROW" on an array when the user selects something in a separate custom component and it successfully updates the data but the UI does not refresh until another change is made such as saving.
